<html>
<head>
<title>Ensembl Rapid Release</title>
</head>
<body>
<script type="application/ld+json">
{
  "@context": "http://schema.org",
  "@type": "WebSite",
  "name": "Ensembl",
  "url": "http://www.ensembl.org/",
  "potentialAction": {
    "@type": "SearchAction",
    "target": "http://www.ensembl.org/Multi/Search/Results?q={search_term_string}",
    "query-input": "required name=search_term_string"
  }
}
</script>

<div class="grid-container">
  <div class="grid-row">
    <div class="grid-col-4">
      [[SCRIPT::EnsEMBL::Web::Document::HTML::ToolBox]]
      <div class="tinted-box no-top-margin home-search">
        [[SCRIPT::EnsEMBL::Web::Document::HTML::HomeSearch]]
      </div>
      <div class="plain-box no-top-margin">
        <p>Ensembl Rapid Release is a new site designed to make our data available more quickly. 
Release of data occurs on a two-week cycle, meaning we can make our gene sets available with 
minimal delay once the annotation is complete. For each species we provide a gene set along with 
additional features such as protein feature annotation and BLAST functionality.</p>

        <p>It is important to note that Ensembl Rapid Release is by nature not as fully featured 
as a typical data release on <a href="https://www.ensembl.org">www.ensembl.org</a>. Currently 
we do not provide data archiving or programmatic access. We are gradually rolling out comparative
analyses across all species, and are working on adding more functionality over the coming months 
to further improve usability.</p>

        <p><a href="/info/about/release_cycle.html">More details</a> about Ensembl Rapid Release
and the current and planned features.</p>
      </div>
      <div class="plain-box no-top-margin">
        <div class="js_panel">
          <input type="hidden" class="panel_type" value="SpeciesList" />
            [[SCRIPT::EnsEMBL::Web::Document::HTML::GenomeList]]
        </div>
      </div>
    </div>
    <div class="grid-col-2">
      <div class="plain-box unbordered whats-new">      		
        [[SCRIPT::EnsEMBL::Web::Document::HTML::WhatsNew]]
        <h2>Comparative analyses</h2>
        <ul>
          <li>All species now have homologues with an appropriate set of reference species,
              allowing us to assign gene symbols where possible</li>
          <li>We are adding <a href="info/data/ftp/index.html">HAL multiple alignments</a> 
            for selected clades, available to download from our FTP site</li>
        </ul>
	<h2> Variation </h2>
        <ul>
	  <li> We display variation data from gnomAD Genomes v3.1.2 and ClinVar 2022-07-09 (lifted over from GRCh38 using crossmap and the <a href="https://github.com/marbl/CHM13"> chain file </a> in the T2T-CHM13v2.0 genome browser. Both variant sets can be downloaded via the <a href="https://ftp.ensembl.org/pub/rapid-release/species/Homo_sapiens/GCA_009914755.4/variation/2022_08/vcf"> FTP </a> in VCFs, with molecular consequences predicted by Ensembl VEP </li>
	  <li> Additionally, we have generated an Ensembl VEP cache containing the remapped Ensembl/GENCODE gene set for T2T-CHM13v2.0 to facilitate analysing your own variant data with Ensembl VEP. This is also available via the <a href="https://ftp.ensembl.org/pub/rapid-release/species/Homo_sapiens/GCA_009914755.4/variation/2022_08/indexed_vep_cache/"> FTP </a>. The gnomAD and ClinVar data described above can be used in VEP analysis using the <a href="http://www.ensembl.org/info/docs/tools/vep/script/vep_custom.html"> custom annotations </a> option.
        </ul>
      </div>
    </div>
  </div>
</div>
  
</body>
</html>
