<html>
<head>
<title>Ensembl Genebuild for non-vertebrate species</title>
</head>

<body>

<h1>Ensembl Genebuild for non-vertebrate species</h1>

<p>This document describes the annotation process of an assembly. The first stage is Assembly Loading where databases are prepared and the assembly loaded into the database.</p>

<h2 id="genome-preparation">Genome preparation</h2>

<p>The genome phase of the Ensembl gene annotation pipeline involves loading an assembly into the 
Ensembl core database schema and then running a series of analyses on the loaded assembly to 
identify an initial set of genomic features. The most important aspect of this phase is 
identifying repeat features, as soft masking of the genome is used extensively later in the 
annotation process.</p>

<h3 id="repeat-finding">Repeat finding</h3>

<p>After the genomic sequence has been loaded into a database, it is screened for sequence 
patterns, including repeats, using Red [<a href="#f1">1</a>], dustmasker [<a href="#f2">2</a>] and TRF [<a href="#f3">3</a>].</p>

<h3 id="low-complexity-features-ab-initio-predictions">Low complexity features, ab initio predictions</h3>

<p>Transcription start sites are predicted using Eponine–scan [<a href="#f4">4</a>]. CpG islands 
longer than 400 bases and tRNAs are also predicted. The results of Eponine-scan, 
CpG [<a href="#f5">5</a>], and tRNAscan [<a href="#f6">6</a>] are for display purposes only; they 
are not used in the gene annotation process.</p>

<h2 id="protein-coding-model-generation">Protein-coding model generation</h2>

<p>Various sources of transcript and protein data are investigated and used to generate gene 
models using a variety of techniques. The data and techniques employed to generate models are 
outlined here.</p>

<h3 id="protein-to-genome-pipeline">Protein-to-genome pipeline</h3>

<p>Protein sequences are downloaded from UniProt and aligned to the genome in a splice-aware 
manner using GenBlast [<a href="#f7">7</a>]. The set of proteins aligned to the genome are 
subsets of UniProt [<a href="#f8">8</a>] proteins and OrthoDB [<a href="#f9">9</a>] proteins 
used to provide a broad, targeted coverage of the species of interest proteome.</p>

<p>A cut-off of 50 percent coverage and 30 percent identity, and an e-value of e-1, were used 
for GenBlast, with the exon repair option turned on. The top 10 transcript models built by 
GenBlast for each protein passing the cut-offs are kept.</p>

<h3 id="rna-seq-pipeline">RNA-seq pipeline</h3>

<p>Where available, RNA-seq data is downloaded from <a href="https://www.ebi.ac.uk/ena/">ENA</a>) 
and used in the annotation. The available reads are aligned to the genome using 
STAR [<a href="#f10">10</a>] and models assembled using Scallop [<a href="#f11">11</a>]. 
Protein-coding models are validated via Diamond [<a href="#f12">12</a>] alignment of the longest 
ORF against a database of eukaryote Uniprot proteins. Where the alignments are insufficient, 
validation is achieved via RNASamba [<a href="#f13">13</a>] and CPC2 [<a href="#f14">14</a>].</p>

<h3 id="long-read-transcriptomic-data-pipeline">Long-read transcriptomic data pipeline</h3>

<p>Where available, long-read transcriptomic data (i.e., IsoSeq or Nanopore) is downloaded from 
ENA and used in the annotation. The long-read data is mapped to the genome using 
Minimap2 [<a href="#f15">15</a>] with the recommended settings for Iso-Seq and Nanopore data. 
Due to the high error rate of the Nanopore data, post mapping error correction is applied to 
maximize the number of usable mappings. Low frequency intron/exon boundaries that are 
non-canonical are replaced with high frequency boundary coordinates within 50bp. In addition, 
low frequency potential gaps between adjoining exons are filled in based on high frequency 
observations of single exons with the same terminal boundary coordinates.</p>

<h3 id="prioritising-models">Prioritising models at each locus</h3>

<p>Low quality models are removed, and data is collapsed and consolidated into a final set of gene 
models plus their associated non-redundant transcripts. When collapsing the data, priority 
is given to models derived from transcriptomic data; where RNA-seq data is fragmented or missing, 
homology data takes precedence.</p>

<h3 id="addition-of-utr-to-coding-models">Addition of UTR to coding models</h3>

<p>The set of coding models is extended into the untranslated regions (UTRs) using RNA-seq data 
(if available). The criterion for adding UTR from RNA-seq alignments to protein models lacking 
UTR (such as the protein-to-genome alignment models) is that the intron coordinates from the model 
missing UTR exactly match a subset of the coordinates from the UTR donor model.</p>

<h3 id="stable-identifiers">Stable identifiers</h3>

<p>Stable identifiers are assigned to each gene, transcript, exon and translation. When annotating 
a species for the first time, these identifiers are auto-generated. In all subsequent annotations 
for a species, the stable identifiers are propagated based on comparison of the new gene set to 
the previous gene set.</p>

<h2 id="appendix-further-information">Appendix - Further information</h2>

<p>The Ensembl gene set is generated automatically, meaning that gene models are annotated using 
the Ensembl gene annotation pipelines. The main focus of the pipelines is to generate a 
conservative set of protein-coding gene models.</p>

<p>Every gene model produced by the Ensembl gene annotation pipeline is supported by biological 
sequence evidence (see the “Supporting evidence” link on the left-hand menu of a Gene page or 
Transcript page); ab initio models are not included in our gene set.</p>

<h2 id="references">References</h2>

<ol>
<li id="f1"><b>Girgis, H.Z.</b> Red: an intelligent, rapid, accurate tool for detecting repeats 
de-novo on the genomic scale. <i>BMC Bioinformatics 16</i>, 227 (2015).</li>
<li id="f2"><b>Morgulis, A., et al.</b>, A fast and symmetric DUST implementation to mask 
low-complexity DNA sequences. <i>J Comput Biol</i>, 2006. 13(5): p. 1028-40.</li>
<li id="f3"><b>Benson, G.</b>, Tandem repeats finder: a program to analyze DNA sequences. <i>Nucleic Acids Res</i>, 1999. 27(2): p. 573-80.</li>
<li id="f4"><b>Down, T.A. and T.J. Hubbard</b>, Computational detection and location of transcription start sites in mammalian genomic DNA. <i>Genome Res</i>, 2002. 12(3): p. 458-61.</li>
<li id="f5"><b>Larsen, F., et al.</b>, CpG islands as gene markers in the human genome. <i>Genomics</i>, 1992. 13(4): p. 1095-107.</li>
<li id="f6"><b>Lowe, T.M. and S.R. Eddy</b>, tRNAscan-SE: a program for improved detection of transfer RNA genes in genomic sequence. <i>Nucleic Acids Res</i>, 1997. 25(5): p. 955-64.</li>
<li id="f7"><b>Li, H. and R. Durbin</b>, Fast and accurate short read alignment with Burrows-Wheeler transform. <i>Bioinformatics</i>, 2009. 25(14): p. 1754-60.</li>
<li id="f8"><b>UniProt Consortium, T.</b>, UniProt: the universal protein knowledgebase. <i>Nucleic Acids Res</i>, 2018. 46(5): p. 2699.</li>
<li id="f9"><b>Kriventseva E.K., et al.</b> OrthoDB v10: sampling the diversity of animal, plant, fungal, protist, bacterial and viral genomes for evolutionary and functional annotations of orthologs , <i>Nucleic Acids Res</i>, Nov 2018</li>
<li id="f10"><b>Alexander Dobin, Carrie A. Davis, Felix Schlesinger, Jorg Drenkow, Chris Zaleski, Sonali Jha, Philippe Batut, Mark Chaisson, Thomas R. Gingeras</b>, STAR: ultrafast universal RNA-seq aligner, <i>Bioinformatics</i>, Volume 29, Issue 1, January 2013, Pages 15–21, <a href="https://doi.org/10.1093/bioinformatics/bts635">doi.org/10.1093</a></li>
<li id="f11"><b>Shao, M., Kingsford, C.</b> Accurate assembly of transcripts through phase-preserving graph decomposition. <i>Nat Biotechnol</i> 35, 1167–1169 (2017). <a href="https://doi.org/10.1038/nbt">doi.org/10.1038</a>.</li>
<li id="f12"><b>Buchfink, B., Reuter, K., Drost, HG.</b> Sensitive protein alignments at tree-of-life scale using DIAMOND. <i>Nature Methods</i> 18, 366–368 (2021)</li>
<li id="f13"><b>Camargo, A. P., Sourkov, V., Pereira, G. A. G. & Carazzolle, M. F.</b> RNAsamba: neural network-based assessment of the protein-coding potential of RNA sequences. <i>NAR Genomics and Bioinformatics</i> 2, lqz024 (2020).</li>
<li id="f14"><b>Kang Y. J., Yang D. C., Kong L., Hou M., Meng Y. Q., Wei L., Gao G.</b> CPC2: a fast and accurate coding potential calculator based on sequence intrinsic features. <i>Nucleic Acids Research</i> 45 (Web Server issue): W12–W16 (2017).</li>
<li id="f15"><b>Li, H.</b> Minimap2: pairwise alignment for nucleotide sequences. <i>Bioinformatics</i>, 2018. 34:3094-3100.</li>
</ol>

</body>
</html>
